beeai/mcp_server/gitlab_tools.py

```diff
@@ -67,14 +67,20 @@ async def open_merge_request(
             raise
     if not pr:
         return "Failed to open the merge request"
-    # by default, set this label on a newly created MR so we can inspect it ASAP
-    try:
-        # Add a short delay before adding the label to avoid race conditions with MR creation
-        await asyncio.sleep(0.5)
-        await asyncio.to_thread(pr.add_label, "jotnar_needs_attention")
-    except OgrException as ex:
-        logger.error("Unable to set label 'jotnar_needs_attention' on MR %s (%s)", pr, ex)
-        # we should still continue and return the MR URL
+
+    # First, verify the MR exists (by checking pr.status) before trying to add the label
+    for attempt in range(5):
+        try:
+            _ = pr.status
```

Comment:
The only thing I can think of that would prevent caching here is:

```suggestion
            _ = await asyncio.to_thread(project.get_pr, pr.id)
```

It should fail when the MR is not found.